kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java

```diff
@@ -55,179 +55,286 @@ public UserTaskAdminServicesClientImpl(KieServicesConfiguration config, ClassLoa
             ServiceResponse<?> response = (ServiceResponse<?>) executeJmsCommand( script, DescriptorCommand.class.getName(), "BPM", containerId ).getResponses().get(0);
             throwExceptionOnFailure(response);
         }
     }
 
     @Override
     public void addBusinessAdmins(String containerId, Long taskId, boolean removeExisting, OrgEntities orgEntities) {
+        addBusinessAdmins(null, containerId, taskId, removeExisting, orgEntities);
+    }
+    @Override
+    public void addBusinessAdmins(String userId, String containerId, Long taskId, boolean removeExisting, OrgEntities orgEntities) {
         if( config.isRest() ) {
             Map<String, Object> valuesMap = new HashMap<String, Object>();
             valuesMap.put(CONTAINER_ID, containerId);
             valuesMap.put(TASK_INSTANCE_ID, taskId);
 
             Map<String, String> headers = new HashMap<String, String>();
             String queryString = "?remove=" + removeExisting;
-
+            if(BYPASS_AUTH_USER && userId != null) {
+                queryString += "&user=" + userId;
+            }
             makeHttpPutRequestAndCreateCustomResponse(
                     build(loadBalancer.getUrl(), ADMIN_TASK_URI + "/" + TASK_INSTANCE_ADMINS_USERS_URI, valuesMap) + queryString, orgEntities, null, headers);
         } else {
+            Object[] argvs = new Object[]{containerId, taskId, removeExisting};
+            if(BYPASS_AUTH_USER && userId != null) {
+                argvs = new Object[]{userId, containerId, taskId, removeExisting};
+            }
             CommandScript script = new CommandScript( Collections.singletonList(
-                    (KieServerCommand) new DescriptorCommand("UserTaskAdminService", "addBusinessAdmins", serialize(orgEntities), marshaller.getFormat().getType(), new Object[]{containerId, taskId, removeExisting})));
+                    (KieServerCommand) new DescriptorCommand("UserTaskAdminService", "addBusinessAdmins", serialize(orgEntities), marshaller.getFormat().getType(), argvs)));
             ServiceResponse<?> response = (ServiceResponse<?>) executeJmsCommand( script, DescriptorCommand.class.getName(), "BPM", containerId ).getResponses().get(0);
             throwExceptionOnFailure(response);
         }
     }
 
     @Override
     public void removePotentialOwnerUsers(String containerId, Long taskId, String... users) {
+        removePotentialOwnerUsers(null, containerId, taskId, users);
+    }
+    @Override
+    public void removePotentialOwnerUsers(String userId, String containerId, Long taskId, String... users) {
 
         if( config.isRest() ) {
             Map<String, Object> valuesMap = new HashMap<String, Object>();
             valuesMap.put(CONTAINER_ID, containerId);
             valuesMap.put(TASK_INSTANCE_ID, taskId);
             valuesMap.put(ENTITY_ID, Arrays.stream(users).collect(Collectors.joining(",")));
 
+            String queryString = "";
```

Comment:
https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/AbstractKieServicesClientImpl.java#L696 can be reused here

Reply:
done